src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Doctrine\Common\Persistence\ObjectManager;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Sylius\Component\Core\Test\Factory\TestPromotionFactoryInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Promotion\Factory\ActionFactoryInterface;
+use Sylius\Component\Promotion\Model\ActionInterface;
+use Sylius\Component\Promotion\Model\RuleInterface;
+use Sylius\Component\Promotion\Repository\PromotionRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class PromotionContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var ActionFactoryInterface
+     */
+    private $actionFactory;
+
+    /**
+     * @var TestPromotionFactoryInterface
+     */
+    private $testPromotionFactory;
+
+    /**
+     * @var PromotionRepositoryInterface
+     */
+    private $promotionRepository;
+
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param ActionFactoryInterface $actionFactory
+     * @param TestPromotionFactoryInterface $testPromotionFactory
+     * @param PromotionRepositoryInterface $promotionRepository
+     * @param ObjectManager $objectManager
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        ActionFactoryInterface $actionFactory,
+        TestPromotionFactoryInterface $testPromotionFactory,
+        PromotionRepositoryInterface $promotionRepository,
+        ObjectManager $objectManager
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->actionFactory = $actionFactory;
+        $this->testPromotionFactory = $testPromotionFactory;
+        $this->promotionRepository = $promotionRepository;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @Given there is a promotion :promotionName
+     */
+    public function thereIsPromotion($promotionName)
+    {
+        $promotion = $this->testPromotionFactory->create($promotionName);
+        $promotion->addChannel($this->sharedStorage->get('channel'));
```

Comment:
How about `$this->testPromotionFactory->createForChannel($promotionName, $this->sharedStorage->get('channel'));`?